src/Tags/Structure.php

```diff
@@ -123,11 +125,18 @@ public function toArray($tree, $parent = null, $depth = 1)
         $pages = collect($tree)->map(function ($item, $index) use ($parent, $depth, $tree) {
             $page = $item['page'];
             $keys = $this->getQuerySelectKeys($page);
-            $data = $page->toAugmentedArray($keys);
+
+            $ref = $page->id() ?? $page->reference();
```

Comment:
Some users may see performance improvements if these two are the other way around:

```suggestion
            $ref = $page->reference() ?? $page->id();
```